app/models/link.rb

```diff
@@ -1,4 +1,6 @@
+require 'uri'
+
 class Link < ActiveRecord::Base
   validates :shortlink, presence: true, uniqueness: true
-  validates :url, presence: true
+  validates :url, presence: true, format: { with: URI::regexp(%w(http https)) }
```

Comment:
Make sure this doesn't leave a vulnerability where you can do payloads like `javascript:alert(1) http://` to circumvent this validation.